android/src/main/java/com/marigold/rnsdk/RNMarigoldPackage.kt

```diff
@@ -0,0 +1,115 @@
+package com.marigold.rnsdk
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.ReactPackage
+import com.facebook.react.bridge.JavaScriptModule
+import com.facebook.react.bridge.NativeModule
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.uimanager.ViewManager
+import com.marigold.sdk.Marigold
+
+class RNMarigoldPackage : ReactPackage {
+
+    var displayInAppNotifications = true
+
+    @VisibleForTesting
+    private val marigold = Marigold()
+
+    /**
+     * Default constructor - for auto linking
+     */
+    constructor() {
+    }
+
+    constructor(context: Context, appKey: String) {
+        marigold.startEngine(context, appKey)
+    }
+
+    /**
+     * Constructor for the RNMarigoldPackage.
+     *
+     * @param context                   the application context
+     * @param appKey                    the app key provided when you registered your application.
+     */
+    constructor(builder: Builder) {
+        marigold.setGeoIpTrackingDefault(builder.geoIPTrackingDefault)
+        marigold.startEngine(builder.context, builder.appKey)
+        displayInAppNotifications = builder.displayInAppNotifications
+    }
+
+    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
+        val modules: MutableList<NativeModule> = ArrayList()
+        modules.add(RNMarigoldModule(reactContext, displayInAppNotifications))
```

Comment:
We'll need to add the new `EngageBySailthruModule` here

Reply:
The `RNEngageBySailthruModule` module still needs to be added here (also to the tests).